cirq/circuits/text_diagram_drawer.py

```diff
@@ -305,3 +330,124 @@ def render(self,
             block_diagram.mutable_block(x, y).content = v.text
 
         return block_diagram.render()
+
+    def copy(self):
+        return self.__class__(entries=self.entries,
+                              vertical_lines=self.vertical_lines,
+                              horizontal_lines=self.horizontal_lines,
+                              vertical_padding=self.vertical_padding,
+                              horizontal_padding=self.horizontal_padding)
+
+    def shift(self, dx: int = 0, dy: int = 0) -> 'TextDiagramDrawer':
+        self._transform_coordinates(lambda x, y: (x + dx, y + dy))
+        return self
+
+    def shifted(self, dx: int = 0, dy: int = 0) -> 'TextDiagramDrawer':
+        return self.copy().shift(dx, dy)
+
+    def update(self, other: 'TextDiagramDrawer') -> 'TextDiagramDrawer':
```

Comment:
This method needs a better name. It overwrites entries but adds lines? Looks like `draw_on_top_of` or similar.

Reply:
How about `superimpose`? In any case, it would be good for the name to indicate (as I think `update` does) that the changes are made in place. So maybe `superimpose_inplace` or `inplace_superimpose`?

Reply:
Just `superimpose` is great. You can add `superimposed` as a variant that operates on a copy, similar to `shift` vs `shifted`.